synapse/config/server.py

```diff
@@ -122,6 +122,30 @@ def read_config(self, config):
             for domain in federation_domain_whitelist:
                 self.federation_domain_whitelist[domain] = True
 
+        self.federation_ip_range_blacklist = config.get(
+            "federation_ip_range_blacklist", None,
```

Comment:
If we make the default `[]` rather than `None` we can save ourselves a bunch of special-casing?

Reply:
I suppose so since this is a blacklist and an empty one is the same as not having one at all.